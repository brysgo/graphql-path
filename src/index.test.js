import gql from "./";
import { print } from "graphql/language";


describe("graphqlPath", () => {
  it("returns a map of paths in a graphql query", () => {
    const { parsedQuery: fooFragment } = gql`
      fragment Foo on Bar {
        blah
      }
    `;
    const { parsedQuery, fragmentNames, fragmentPaths } = gql`
      query FooQuery {
        ${"somethingOnRoot"}
        someResource {
          ${"onSomeResource"}
          anotherField {
            ${"onAnotherField"}
            ...Foo
          }
        }
      }

      ${fooFragment}
    `;
    expect(print(parsedQuery)).toMatchSnapshot();
    expect(fragmentNames.get(fooFragment)).toEqual("Foo");
    expect(fragmentPaths).toEqual({
      somethingOnRoot: "",
      onSomeResource: "someResource",
      onAnotherField: "someResource.anotherField",
      Foo: "someResource.anotherField"
    });
  });
});
