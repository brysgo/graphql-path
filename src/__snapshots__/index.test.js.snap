// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphqlPath returns a map of paths in a graphql query 1`] = `
"query FooQuery {
  somethingOnRoot
  someResource {
    onSomeResource
    anotherField {
      onAnotherField
      ...Foo
    }
  }
}

fragment Foo on Bar {
  blah
}
"
`;
